apps/laboratory/tests/shared/utils/email.ts

```diff
@@ -0,0 +1,60 @@
+import { Mailsac, type EmailMessage } from '@mailsac/api'
+
+export class Email {
+  private readonly mailsac: Mailsac<any>
+  private messageCount: any
+  constructor(public readonly apiKey: string) {
+    this.mailsac = new Mailsac({ headers: { 'Mailsac-Key': apiKey } })
+    this.messageCount = undefined
+  }
+
+  async deleteAllMessages(email: string) {
+    this.messageCount = 0
+    return await this.mailsac.messages.deleteAllMessages(email)
+  }
+
+  async getNewMessage(email: string) {
+    const timeout = new Promise((_, reject) => {
+      setTimeout(() => {
+        reject('timeout')
+      }, 15000)
+    })
+
+    const messagePoll = new Promise(resolve => {
+      const interval = setInterval(async () => {
+        const messages = await this.mailsac.messages.listMessages(email)
```

Comment:
Nice find with this tool!

Looks like we have less than 500 test runs per month in the free plan so it's possible we might have to pay at some point: https://docs.mailsac.com/en/latest/services/api_calls/api_calls.html

Reply:
Just signed up for the paid plan after I didn't see any initial pushback.
Paid plan gives us 25k interactions and 50 reserved emails.

I'm using 10 of these 50 for a naive collission avoidance strategy where each test cleans up the email's mailbox and then runs. Could be more sophisticated about this but works for now.